one-ui/src/main/java/com/marklogic/hub/oneui/models/HubConfigSession.java

```diff
@@ -1002,4 +1008,24 @@ public void afterPropertiesSet() throws Exception {
         hubConfigImpl.hydrateConfigs();
         this.dataHub = new DataHubImpl(this);
     }
+
+    @Override
+    public void destroy() {
+        if (!clientsByKindAndDatabaseName.isEmpty()) {
+            clientsByKindAndDatabaseName.values().stream()
+                .flatMap(s -> s.values().stream().filter(Objects::nonNull))
+                .forEach(
+                    e -> {
+                        logger.debug(String.format("release %s (%s)", e.getDatabase(), e.toString()));
+                        e.release();
+                        e = null;
+                    });
+            clientsByKindAndDatabaseName.clear();
+        }
+    }
+
+    //only for test purpose
+    public Map<DatabaseKind, Map<String, DatabaseClient>> getAllDatabaseClients() {
```

Comment:
This method can be "protected" and test can be added to same package

Reply:
@srinathgit good suggestion...